dsp/Cargo.toml

```diff
@@ -9,6 +9,10 @@ libm = "0.2.1"
 serde = { version = "1.0", features = ["derive"], default-features = false }
 generic-array = "0.14"
 
+[dependencies.miniconf]
```

Comment:
I think this is better done with a `patch` at the root. `dsp` shouldn't need `miniconf`, right?

Reply:
dsp needs `miniconf::StringSet` so that it can use it in the `#[derive(StringSet)]` for the DSP types, like IIR. If we don't want those to be directly settable, we can remove the dependency

Reply:
Ah ok.
I was wondering when it would make sense to bring miniconf down into dsp. And this is just that case. Otherwise, we'd have to duplicate the IIR struct (structurally) in the settings.

Reply:
But I still think the this should be done with a `patch` at the root Cargo.toml, no?

Reply:
`patch` is generally used to override an existing dependency with a different version. In this case, `miniconf` isn't available at all in our dependency tree, so I'm not sure what we could `patch`. Unless you're referring to something else?

I dislike having to doubly specify the `miniconf` dependency, but I think that's a result of how we've structed `dsp` as a separate crate instead of a module (like `hardware`) within the `stabilizer` crate.

Reply:
Unless you're specifying the specify `branch` specifier to only appear at the root and use patch to push it down our dependency graph?

I plan on waiting until that branch is merged to mainline and we release miniconf before merging this, so I've been leaving it as-is for now.

Reply:
```
[dependencies]
miniconf = "0.1"
```

in both Cargo.toml and then

```
[patch.crates-io]
miniconf = { git = "https://github.com/quartiq/miniconf.git", branch = "rs/issue-21/terminal-array-elements"}
```

only at the root works afaict.

Reply:
Yeah. It's a nit. Just a few fewer lines.